src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -4881,7 +4870,8 @@ public static MethodHandle identity(Class<?> type) {
      */
     public static MethodHandle zero(Class<?> type) {
         Objects.requireNonNull(type);
-        return type.isPrimitive() ?  zero(Wrapper.forPrimitiveType(type), type) : zero(Wrapper.OBJECT, type);
+        return type.isPrimitive() ? primitiveZero(Wrapper.forPrimitiveType(type))
+                : MethodHandleImpl.makeConstantReturning(type, null);
```

Comment:
Not sure if it's important but the existing impl would cache `zero(Object.class)` while this new impl won't. Behaviorally neutral for any other reference type, though.

Reply:
`type == Object.class` can use the primitive path, but I am too lazy to add another of this check.